target/linux/ramips/mt7621/base-files/lib/upgrade/iodata.sh

```diff
@@ -16,24 +16,28 @@ iodata_mstc_prepare_fail() {
 #     use 1st image in OpenWrt
 # - debugflag: enable/disable debug
 #     users can interrupt Z-Loader for recovering the device if enabled
+#
+# parameters:
+# - $1: the offset of "debugflag"
 iodata_mstc_upgrade_prepare() {
 	local persist_mtd="$(find_mtd_part persist)"
 	local factory_mtd="$(find_mtd_part factory)"
+	local dflag_offset="${1}"
 
 	if [ -z "$persist_mtd" -o -z "$factory_mtd" ]; then
 		echo 'cannot find mtd partition(s), "factory" or "persist"'
 		iodata_mstc_prepare_fail
 	fi
 
 	local bootnum=$(hexdump -s 4 -n 1 -e '"%x"' ${persist_mtd})
-	local debugflag=$(hexdump -s 65141 -n 1 -e '"%x"' ${factory_mtd})
+	local debugflag=$(hexdump -s $((${dflag_offset})) -n 1 -e '"%x"' ${factory_mtd})
```

Comment:
You can drop the inner "$", just `$((dflag_offset))` is enough.

Reply:
OK, when I was writing it, I just intended to emphasize that it was a variable. I'll update.